File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/typescript/F14.P2.ts
interface Circle { kind: "Circle"; r: number }
interface Rect { kind: "Rect"; w: number; h: number }
type Shape = Circle | Rect

function areaOf(s: Shape): number {
// BEGIN PROBE F14.P2
let area: number
switch (s.kind) {
case "Circle":
area = 3.141592653589793 * s.r * s.r
break
case "Rect":
area = s.w * s.h
break
}
return area
// END PROBE F14.P2
}

console.log(areaOf({ kind: "Circle", r: 2.0 }))

File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/go/F14.P2.go
package main

import "fmt"

type Shape interface{ isShape() }

type Circle struct{ r float64 }

func (Circle) isShape() {}

type Rect struct{ w, h float64 }

func (Rect) isShape() {}

func areaOf(s Shape) float64 {
	// BEGIN PROBE F14.P2
	var area float64
	switch v := s.(type) {
	case Circle:
		area = 3.141592653589793 * v.r * v.r
	case Rect:
		area = v.w * v.h
	}
	return area
	// END PROBE F14.P2
}

func main() {
	fmt.Println(areaOf(Circle{2.0}))
}
